src/transformers/models/yolos/configuration_yolos.py

```diff
@@ -75,7 +75,7 @@ class YolosConfig(PretrainedConfig):
         use_mid_position_embeddings (`bool`, *optional*, defaults to `True`):
             Whether to use the mid-layer position encodings.
         auxiliary_loss (`bool`, *optional*, defaults to `False`):
-            Whether auxiliary decoding losses (loss at each decoder layer) are to be used.
+            Whether auxiliary decoding losses (loss at each decoder layer) are to be used. Due to the incompatibility with auxiliary_loss, please refrain from altering this variable..
```

Comment:
Could you explain this comment a bit? [config.auxiliary_loss is used in YolosForObjectDetection](https://github.com/huggingface/transformers/blob/2382706a1ca038099214664dfa40f2f06883072c/src/transformers/models/yolos/modeling_yolos.py#L815) so if something's incompatible we'll need to update yolos' modeling file

Reply:
@amyeroberts https://github.com/huggingface/transformers/blob/2382706a1ca038099214664dfa40f2f06883072c/src/transformers/models/yolos/modeling_yolos.py#L666
As you see in the code and original YOLOS, they don't have function for `outputs.intermediate_hidden_states`

Reply:
So I thought it should be eliminated at the first place but I just remain it, as you suggested!

Reply:
Just to make sure we're on the same page. Yolos does use this flag [here](https://github.com/huggingface/transformers/blob/2382706a1ca038099214664dfa40f2f06883072c/src/transformers/models/yolos/modeling_yolos.py#L815). 

So either we should remove the additional warning here (users should be allow to modify this parameter) or we need to fix something in yolos

Reply:
@amyeroberts 
> So either we should remove the additional warning here (users should be allow to modify this parameter) or we need to fix something in yolos

Yeah I also agree, but I think I need to explain in very detail.

Transformers version of YOLOS has the flag of `config.auxiliary_loss`. However, the original code of yolos (which i recommend you to look at [line 61](https://github.com/hustvl/YOLOS/blob/5717fc29d727dab84ad585c56457b4de1225eddc/models/detector.py#L61), and [line 815](https://github.com/hustvl/YOLOS/blob/5717fc29d727dab84ad585c56457b4de1225eddc/models/detector.py#L227)) also has an same error that auxiliary_output does not exist at the forward path. So I assume that when we use original code of yolos with enabling auxiliary_loss it will return error because `outputs['aux_outputs']` does not have variable `aux_outputs`. **Decisively [Yolos](https://github.com/hustvl/YOLOS/blob/main/main.py) does not have enabling configuration related to auxiliary_loss.**

So the original modification (which i reverted) was to get rid of the flag of `config.auxiliary_loss` since Yolos is not compatible for this variable. However, as you mentioned earlier we should be able to keep reuse the original configuration and also backward compatibility I just remained as original and just added caution.

Now, are we on the same page?

Reply:
@SangbumChoi Thanks for your detailed reply! 

Even if this isn't possible in the original Yolos, the question of how to handle the argument here is whether it's possible in the implementation in this repo., which I understand that it is. It's OK for additional functionality to be possible, which wasn't in the original model, provided it's "correct" (which is arguably hard to verify here because of the original implementation). 

I'd recommend removing the warning here to keep the model behaviour consistent.

Reply:
Thanks for reply!
> It's OK for additional functionality to be possible, which wasn't in the original model, provided it's "correct" (which is arguably hard to verify here because of the original implementation).

Okay, let's just keep in mind that for the case of yolos enabling config.auxiliary_loss will not acting same behavior as other transformers!